src/Lang/Cpp/CppDocGen.ts

```diff
@@ -228,6 +228,34 @@ export class CppDocGen implements IDocGen {
         }
     }
 
+    protected generateVersionTag(lines: string[]) {
+        if (this.cfg.File.versionTag.trim().length !== 0) {
+            lines.push(this.cfg.C.commentPrefix + this.cfg.File.versionTag);
+        }
+    }
+
+    protected generateCopyrightTag(lines: string[]) {
+        this.cfg.File.copyrightTag.forEach((element) => {
+            this.generateFromTemplate(
+                lines,
+                this.cfg.yearTemplateReplace,
+                element,
+                [moment().format("YYYY")],
+            );
+        });
+    }
+
+    protected generateCustomTag(lines: string[]) {
+        this.cfg.File.customTag.forEach((element) => {
+            this.generateFromTemplate(
+                lines,
+                this.cfg.dateTemplateReplace,
```

Comment:
Please check if this isn't set to an empty string.

Reply:
I want to allow empty lines in the customTag.  If the array is completely empty, the for loop shouldn't have any iterations.  For instance, the following should be a valid block to add:
```json
[ "@par First Paragraph",
"First Paragraph Text",
""
"@par Second Paragraph",
"Second Paragraph Text"
]

Reply:
Sorry I wasn't clear on that.
I wanted you to check if `this.cfg.dateTemplateReplace` was empty as it can be set by the user and then the replacement would fuck up. Take a look at line #260
Your solution otherwise is fine.

Reply:
Ah, of course.  That makes sense.